src/Pirouette/Transformations/DCE.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Pirouette.Transformations.DCE where
```

Comment:
Maybe we should really call this module "RemoveDeadConstructors". It will be re-exported by `Pirouette.Transformations` anyway.

Reply:
I was thinking about also putting other dead code removal stuff in this module (and the `C` in `DCE` being for the general "code"), but I can rename! I don't really think we'll be doing dead functions removing anyway — isn't symbolic evaluation effectively oblivious to the functions that are unused?

Reply:
Symbolic evaluation is oblivious to unused functions indeed (at least when doing call-by-name execution)